app/models/user.rb

```diff
@@ -16,7 +16,7 @@ class User < ApplicationRecord
   has_many   :miq_widget_sets, :as => :owner, :dependent => :destroy
   has_many   :miq_reports, :dependent => :nullify
   has_many   :service_orders, :dependent => :nullify
-  has_many   :shares
+  has_many   :owned_shares, :class_name => "Share"
```

Comment:
`user.owned_shares.grep(DesiredClass)` <= I didn't even know you could do that. Gross.

Anything you'd need you should be able to do with normal scopes and queries. 

```
# Shares by type of share
user.owned_shares.where(:resource_type => DesiredClass)

# I just added this association in a new commit to demo, check it out
# Shares connected to a particular resource
resource.connected_shares
```